### PR TITLE
Fix AudioInjector::restart() for the non-local case.

### PR DESCRIPTION
**Pull request by howard-stearns**
E.g.,:
var HIFI_PUBLIC_BUCKET = "http://s3.amazonaws.com/hifi-public/";
var elevatorSound = SoundCache.getSound(HIFI_PUBLIC_BUCKET + "sounds/Lobby/elevator.stereo.raw");
var elevatorInjector = Audio.playSound(elevatorSound, { localOnly: false, position: MyAvatar.position, volume: 0.5 });
elevatorInjector.stop();
elevatorInjector.restart();  // This wasn't working when localOnly: false, above.

The polling loop is to deal with this case (after the above is fixed by this pr but without the polling):
elevatorInjector.restart();
elevatorInjector.restart();  // This was leaving the first injector, which you wouldn't notice until...
elevatorInjector.stop(); // ..this would continue playing the first "paused" injector.
elevatorInjector.stop(); // This was necessary to finally stop both sounds.

### CODE REVIEW

File: libraries/audio/src/AudioInjector.cpp [hunk 1]
```diff
@@ -271,6 +272,14 @@ void AudioInjector::stop() {
...
         // we're only a local injector, so we can say we are finished right away too
         setIsFinished(true);
+    } else {
+        // Wait for actual stop (because synchronous callers such as restart might immediately change state before
+        // the asynchronous injector has a chance to see that it should stop). This prevents issues where, e.g.,
+        // two stop()s are required after two restart()s.
+        while (!_isFinished) {
+            //qCDebug(audio) << AudioConstants::NETWORK_FRAME_USECS << AudioConstants::NETWORK_FRAME_MSECS;
+            usleep(AudioConstants::NETWORK_FRAME_USECS);
+        }
```
Comment (birarda) [inline libraries/audio/src/AudioInjector.cpp]: is there a way we can handle this with a signal instead of sleeping over and over

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6668de56cd72564b08790ba66db763d73a16551f available at: https://gist.github.com/fb9f8411cf007eb449e7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.dmg

File: libraries/audio/src/AudioInjector.cpp (hunk 1 above)
Comment (howard-stearns) [inline libraries/audio/src/AudioInjector.cpp]: Maybe.
As I understand the overall design, though, we're really counting on stop() being synchronous.  So a condition-variable pattern would work. (E.g., http://en.cppreference.com/w/cpp/thread/condition_variable). Alas, it seems like qt considers this an anti-pattern, and I'm not (yet) seeing how to do this. Am I missing something?

The good news, though, is that stopping is far less frequent than the actual injector loop, and so this is going to happen much less often than the existing injector usleep(), and I don't think it should occur more than twice in each stop().

Comment (howard-stearns) [inline libraries/audio/src/AudioInjector.cpp]: Oh, duh. Let me take a look at http://doc.qt.io/qt-5.4/qwaitcondition.html

**Comment by hifi-gustavo** [comment]
Build console output for commit: 564b0b48ef53c8243836f48865a7967c5970bf58 available at: https://gist.github.com/fb9f8411cf007eb449e7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.dmg

**Comment by howard-stearns** [comment]
(I've pull from upstream and merged, to make it easier to merge. That's what's building now.)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8e41c58fadcae1ec8878910fe51f0adf0b958698 available at: https://gist.github.com/fb9f8411cf007eb449e7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d02fac3db3e0862a10339bf28e78c09e0431322a available at: https://gist.github.com/fb9f8411cf007eb449e7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: d02fac3db3e0862a10339bf28e78c09e0431322a available at: https://gist.github.com/fb9f8411cf007eb449e7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5074.dmg

**Pull request merged**